src/project-converter.ts

```diff
@@ -0,0 +1,165 @@
+  }
+
+  /**
+   * Specifically convert an HTML document to a JS module. Useful during
+   * conversion for dependencies where the type of result is explictly expected.
+   */
+  convertDocumentToJs(document: Document, visited: Set<OriginalDocumentUrl>) {
+    if (!this.shouldConvertDocument(document)) {
+      return;
+    }
+    const documentConverter = new DocumentConverter(this, document, visited);
+    const newModule = documentConverter.convertToJsModule();
+    this._handleConversionResult(newModule);
+  }
+
+  /**
+   * Specifically convert an HTML document without changing the file type
+   * (changes imports and inline scripts to modules as necessary.) Useful during
+   * conversion for dependencies where the type of result is explictly expected.
+   */
+  convertDocumentToHtml(document: Document, visited: Set<OriginalDocumentUrl>) {
+    if (!this.shouldConvertDocument(document)) {
+      return;
+    }
+    const documentConverter = new DocumentConverter(this, document, visited);
+    const newModule = documentConverter.convertAsToplevelHtmlDocument();
+    this._handleConversionResult(newModule);
+  }
+
+  /**
+   * A private instance method for handling new conversion results, exports,
+   * etc.
+   */
+  private _handleConversionResult(newModule: ConversionResult): void {
+    this.conversionResults.set(newModule.originalUrl, newModule);
+    if (newModule.output.type === 'js-module') {
+      for (const expr of newModule.output.exportedNamespaceMembers) {
+        this.namespacedExports.set(
+            expr.oldNamespacedName,
+            new JsExport(newModule.convertedUrl, expr.es6ExportName));
+      }
+    }
+  }
+
+  /**
+   * This method collects the results after all documents are converted. It
+   * handles out some broken edge-cases (ex: shadycss) and sets empty map
+   * entries for files to be deleted.
+   */
+  getResults(): Map<ConvertedDocumentFilePath, string|undefined> {
+    const results = new Map<ConvertedDocumentFilePath, string|undefined>();
+
+    for (const convertedModule of this.conversionResults.values()) {
+      // These files were already ES6 modules, so don't write our conversions.
+      // Can't be handled in `excludes` because we want to preserve imports of.
+      // TODO(fks): This is hacky, ProjectConverter isn't supposed to know about
+      //  project layout / file location. Move into URLHandler, potentially make
+      //  its own `excludes`-like settings option.
+      if (convertedModule.originalUrl ===
+              'shadycss/entrypoints/apply-shim.js' ||
+          convertedModule.originalUrl ===
+              'bower_components/shadycss/entrypoints/apply-shim.js' ||
+          convertedModule.originalUrl ===
+              'shadycss/entrypoints/custom-style-interface.js' ||
+          convertedModule.originalUrl ===
+              'bower_components/shadycss/entrypoints/custom-style-interface.js') {
+        continue;
+      }
+      if (convertedModule.keepOriginal !== true) {
```

Comment:
why not `=== false` or just `!` if this field is always a boolean? If `undefined` is a valid value and means to keep, then this property should be renamed to `deleteOriginal` where `undefined` means false.

Reply:
Yea normally I would just make this property explictly required, but this option only makes sense if the file is not being overwritten.

Will rename to deleteOriginal